Dev simplified onboarding pull request

Nikunjsharma0 opened the pull request:
- signup with passkey added
- simplified on-boarding screens added
-  add child profile added

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/AddChildProfileViewModel.kt:
```diff
@@ -0,0 +1,235 @@
+    val onAddChildProfile: String? = null,
+    val childProfiles: List<Person> = emptyList(),
+    val personParenJoinList: List<PersonParentJoin> = emptyList(),
+    val profileIncludingParentAndChildren: List<Person> = emptyList(),
```
This looks lke we would be storing the Person object for the child twice; we don't want to do that because it leads to potential inconsistency. 

Why do we need to have the parent person in the uistate? That is already going to be the active account on accountmanager.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/AddChildProfileViewModel.kt:
```diff
@@ -0,0 +1,235 @@
+    val showProfileSelectionDialog: Boolean = false
+)
+
+class AddChildProfileViewModel(
```
Should be AddChildProfilesViewModel (plural)

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/AddOrUpdatedChildprofileuseCase.kt:
```diff
@@ -0,0 +1,29 @@
+
+import com.ustadmobile.lib.db.entities.Person
+
+class AddOrUpdatedChildProfileUseCase {
```
I think this is definitely not needed. We had a special use case for course blocks because we need to set the order field and indentation. We are not saving the child profiles in any special order.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/AddChildProfileViewModel.kt:
```diff
@@ -0,0 +1,235 @@
+                    serializer = ListSerializer(Person.serializer()),
+                    loadFromStateKeys = listOf(STATE_KEY_PERSONS),
+                    onLoadFromDb = { db ->
+                        val ChildProfilesDb = db.personDao()
```
I don't think we need to load this from the database. In this screen the initial list is always empty.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/EditChildProfileViewModel.kt:
```diff
@@ -0,0 +1,177 @@
+                hideBottomNavigation = true,
+            )
+        }
+        _uiState.update { prev ->
```
This uistate.update is not needed - that should be handled by loadentity

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/EditChildProfileViewModel.kt:
```diff
@@ -0,0 +1,177 @@
+            )
+        }
+        viewModelScope.launch {
+            val selectedContentEntry = savedStateHandle.getJson(
```
inaccurate variable name - it is not a selected content entry. This variable does not need to exist.

When navigating from AddChildProfileViewModel - put the JSON of the current value in json in the arguments using the key UstadEditView.ARG_ENTITY_JSON . LoadEntity will look for that by default.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/EditChildProfileViewModel.kt:
```diff
@@ -0,0 +1,177 @@
+                serializer = Person.serializer(),
+                makeDefault = {
+                    val newUid = activeDb.doorPrimaryKeyManager.nextIdAsync(Person.TABLE_ID)
+                    Person(
```
This should only create a blank new person. If a previous value was passed in an argument, then makeDefault is not called.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+    val isTeacher: Boolean = false,
+
+    val signupWithPasskey: Boolean = true,
```
This name description needs to be clarified. signup with passkey is a verb to do something, nota state. Maybe passkeySupported or signupWithPasskeyVisible ?

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+    val signupWithPasskey: Boolean = true,
+
+    val showCreatePasskeyPrompt: Boolean? = false,
```
Why do we need this property?

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+    private val genderConfig: GenderConfig by instance()
+
+    private val enqueueSavePictureUseCase: EnqueueSavePictureUseCase by
+    on(LearningSpace(serverUrl)).instance()
```
needs indentation

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+            )
+        }
+        if (savedStateHandle[SIGN_WITH_USERNAME_AND_PASSWORD] == "true") {
```
Why would this need to be passed as an argument? It can be checked anytime by getting the passkey use case via instance or null, and then checking if that is not null

Nikunjsharma0 replied:
i have done this because if passkey is supported and then user selected other option and in then selected signup with username and password then i am passing this argument .

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+    }
+
+
+    fun onParentCheckChanged(check: Boolean) {
```
Should be checked not check - the state is checked or not checked.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+                genderError != null
+    }
+
+    private fun validateUsername(username: String): Boolean {
```
This should be moved out into being a use case e.g. ValidateUsernameUseCase.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+                if (_uiState.value.signupWithPasskey) {
+                    val passkeyCreated = createPasskeyUseCase?.invoke(
+                        CreatePasskeyParams
```
opening bracket on wrong line

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+                            person = savePerson
+                        )
+                    )
+                    passkeyCreated?.let {
```
This will silently fail with no logging at all if the passkeycreated is null. We don't want that. we should show the user something went wrong and log it.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+                }
+
+
+                val personPictureVal = _uiState.value.personPicture
```
This would need to be done before navigating away.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+    fun onClickOtherOption() {
+
+        navController.navigate(OtherSignUpOptionSelectionViewModel.DEST_NAME, emptyMap())
```
This needs to pass all the registration arguments

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+        const val ARG_DATE_OF_BIRTH = "DateOfBirth"
+
+        val REGISTRATION_ARGS_TO_PASS = listOf(
```
This must not be copy/pasted. We should reference the registration_args_to_pass we already have. There should only be one - otehrwise if an argument is included in one, but not the other, that will create a problem. 

Key principle: single point of truth.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+    val usernameError: String? = null,
+
+    val fullName: String? = null,
```
We don't need fullname as an extra variable. Just use the firstNames field on the Person entity. When it is time to save to the database, then split it.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt:
```diff
@@ -0,0 +1,429 @@
+
+    val firstNameError: String? = null,
+
+    val lastNameError: String? = null,
```
This is obsolete now

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/EditChildProfileViewModel.kt:
```diff
@@ -0,0 +1,177 @@
+
+    val firstNameError: String? = null,
+
+    val lastNameError: String? = null,
```
This should be obsolete now

mikedawson committed:
Update 007_002d_user_registration_above_age_13_test_description.md

mikedawson committed:
Update and rename 007_002e_user_registration_personalal_account_can_add_child_profiles to 007_002e_user_registration_personalal_account_can_add_child_profiles.md

Nikunjsharma0 committed:
SignUpEnterUsernamePasswordScreen added

Nikunjsharma0 committed:
Admin can enable/disable registration for users

mikedawson committed:
Update 007_002a_user_registration_above_age_13_test_description.md

Nikunjsharma0 committed:
child session issue

poojaustad committed:
Merge remote-tracking branch 'origin/dev-simplified-onboarding' into dev-simplified-onboarding-maestro-update

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
Merge branch 'refs/heads/primary' into dev-simplified-onboarding-maestro-update

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
changes in simplified on boarding

poojaustad committed:
changes in simplified on boarding

Nikunjsharma0 committed:
web ui created

poojaustad committed:
changes in existing codes

poojaustad committed:
changes in existing codes

mikedawson committed:
Merge pull request #947 from UstadMobile/dev-simplified-onboarding-maestro-update

Dev simplified onboarding maestro update

Nikunjsharma0 committed:
changes in RedirectViewModel

poojaustad committed:
existing tests

poojaustad committed:
existing tests with conditions

Nikunjsharma0 committed:
preset learnning space check placed in NavControllerExt.kt

poojaustad committed:
Cypress existing test changes

poojaustad committed:
Cypress existing test changes

Nikunjsharma0 committed:
changes in web part

poojaustad committed:
Cypress existing test changes

Nikunjsharma0 committed:
unit test added

mikedawson commented on gradle/libs.versions.toml:
```diff
@@ -204,6 +206,9 @@ autolink = { module = "org.nibor.autolink:autolink", version.ref = "autolink" }
 process-phoenix = { module = "com.jakewharton:process-phoenix", version.ref = "process-phoenix"}
 multiplatformsettings = { module = "com.russhwolf:multiplatform-settings", version.ref = "multiplatformsettings"}
 coil = { module = "io.coil-kt:coil", version.ref = "coil"}
+credentials = { module = "androidx.credentials:credentials", version.ref = "credentials"}
```
name should be prefixed with androidx as per others e.g. androidx-credentials

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/domain/passkey/PasskeyResult.kt:
```diff
@@ -0,0 +1,16 @@
+import com.ustadmobile.lib.db.entities.Person
+
+
+data class PasskeyResult(
```
Should be CreatePasskeyResult. Should not be its own file, should be an inner class in the use case e.g. it should be CreatePasskeyUseCase.CreatePasskeyResult not just CreatePasskeyResult.

mikedawson commented on core/src/commonMain/kotlin/com/ustadmobile/core/domain/person/AddNewPersonUseCase.kt:
```diff
@@ -19,6 +19,7 @@ class AddNewPersonUseCase(
         addedByPersonUid: Long = 0,
         createPersonParentApprovalIfMinor: Boolean = true,
         systemPermissions: Long = SystemPermission.PERSON_DEFAULT_PERMISSIONS,
+        accountType: Int = PersonParentJoin.RELATIONSHIP_OTHER,
```
This parameter has a misleading name. It is not account type - it is relationship. Why is it called that?

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/util/passkey/PasskeyRequestJsonUseCase.kt:
```diff
@@ -0,0 +1,94 @@
+import java.util.Base64
+import kotlin.random.Random
+
+object PasskeyRequestJsonUseCase {
```
This does not follow the Coding Style for a domain UseCase. A UseCase should be a class / interface with an invoke function, not an object.

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/util/passkey/PasskeyRequestJsonUseCase.kt:
```diff
@@ -0,0 +1,94 @@
+        createPasskeyParams: CreatePasskeyParams
+    ): String {
+        val userId = randomString(16)
+        val challenge = Json.encodeToString(
```
Json should be a constructor parameter (dependency) for the use case. Please do not use the singleton. That is slower and it might not have the correct configuration.

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/util/passkey/PasskeyRequestJsonUseCase.kt:
```diff
@@ -0,0 +1,94 @@
+                    "challenge": "${challengeBase64Encoded}",
+                    "rp": {
+                      "id": "credential-manager-${createPasskeyParams.domainName}",
+                      "name": "Ustad Mobile"
```
This name is hardcoded - the use case should add UstadMobileSystemImpl as a constructor parameter and then use the getString function to get the app name.

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/app/App.kt:
```diff
@@ -136,7 +145,7 @@ fun App(
                 //As per https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#navigationbar
                 var selectedTopLevelItemIndex by remember { mutableIntStateOf(0) }
-                if(useBottomBar) {
+                if(useBottomBar && currentSession?.learningSpace?.isLocal != true && currentSession?.person?.isPersonalAccount !=true) {
```
This line is too long

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/app/AppNavHost.kt:
```diff
@@ -203,6 +225,27 @@ fun AppNavHost(
     }
 
+
+    var passkeyPromptData by remember{ mutableStateOf<PassKeyPromptData?>(null) }
```
The prompt to create a passkey after logging in with a username/password is not tested/used yet - please remove this section for now.

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/login/LoginScreen.kt:
```diff
@@ -59,7 +63,12 @@ fun LoginScreen(
+    onSignInWithPasskey: () -> Unit = {},
 ) {
+
+    val filePickLauncher = rememberUstadFilePickLauncher { result ->
```
This file picker is not used and it should be removed.

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/login/LoginScreen.kt:
```diff
@@ -118,23 +126,21 @@ fun LoginScreen(
-        Spacer(modifier = Modifier.height(8.dp))
-
-        if(uiState.createAccountVisible) {
             OutlinedButton(
```
not used must be removed.

mikedawson commented on lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/login/LoginScreen.kt:
```diff
@@ -118,23 +126,21 @@ fun LoginScreen(
             Text(stringResource(MR.strings.login))
         }
+        if(uiState.isPersonalAccount) {
+            Spacer(modifier = Modifier.height(10.dp))
```
All material user interface spacing is in steps of 8dp not 10dp

mikedawson committed:
Merge pull request #963 from UstadMobile/dev-simplified-onboarding-cypress-existing-test-update

Cypress existing test changes

mikedawson committed:
Merge pull request #961 from UstadMobile/dev-simplified-onboarding-maestro-existing-test-update

Maestro existing tests

mikedawson committed:
Merge branch 'refs/heads/dev-simplified-onboarding' into dev-simplified-onboarding-learningspacemgr

# Conflicts:
#	app-react/src/jsMain/kotlin/UstadJsDi.kt

mikedawson committed:
Add changes/todos for learning space list.

Nikunjsharma0 committed:
pr comment

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
worked on todo points

Nikunjsharma0 committed:
added id

poojaustad committed:
Cypress code update

poojaustad committed:
Cypress code update

poojaustad committed:
Cypress code update

Nikunjsharma0 committed:
dependency issue fixed

Nikunjsharma0 committed:
port issue fixed

Nikunjsharma0 committed:
instance changes

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

mikedawson committed:
Add required door entities to SystemDb.

mikedawson committed:
Change endpoint for SystemDb in UstadApp DI.

nikunjsharma1 committed:
commit

nikunjsharma1 committed:
command added

nikunjsharma1 committed:
delete dir code added

nikunjsharma1 committed:
commit

nikunjsharma1 committed:
readme updated to test passkey

nikunjsharma1 committed:
readme updated to test passkey

nikunjsharma1 committed:
crash fixed

nikunjsharma1 committed:
ustad-appconfig command added

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
error fixed

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
commit

Nikunjsharma0 committed:
system db added in js di

Nikunjsharma0 committed:
system db changes

mikedawson committed:
Fix SystemDb - add ReplicateEntity annotation for LearningSpaceInfo.

Nikunjsharma0 committed:
changing admin person to admin user
changes in GoToLearningSpaceUseCaseJs

Nikunjsharma0 committed:
desktop di added

Nikunjsharma0 committed:
readme updated

Nikunjsharma0 committed:
readme updated

Nikunjsharma0 committed:
install.md updated

mikedawson committed:
Update to use a single command with subcommands for running the server and managing the server.

Nikunjsharma0 committed:
readme updated

Nikunjsharma0 committed:
test server controller updated

Nikunjsharma0 committed:
test server controller updated

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding-learningspacemgr

# Conflicts:
#	app-desktop/src/main/java/com/ustadmobile/port/desktop/DesktopDomainDiModule.kt
#	app-react/src/jsMain/kotlin/UstadJsDi.kt
#	gradle/libs.versions.toml
#	testserver-controller/src/main/java/com/ustadmobile/test/http/TestServerController.kt

mikedawson committed:
Fix ServerAppMain when started with no args to run server by default.
UmRestApplication ktor server set to use jsDevServer by default when running from source.

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding-learningspacemgr

mikedawson committed:
Fixes/improvements for initial setup and learning space creation.

1) Make jsdevserver mode automatic when running from source.
2) Update arguments for creating a learning space to ensure clear differentiation between the system config authentication and the learning space admin password.
3) Documentation updates for learning space creation and setup.

mikedawson committed:
Add app-ktor-server/server.properties to gitignore

mikedawson committed:
Fix server setup procedures for binary server build.

mikedawson committed:
Fix testserver-controller to disable using jsDevServer (e.g. reverse proxy for webpack) when using testserver-controller .

mikedawson committed:
Update cypress and maestro scripts to use learning space parameter name.
Add comment on UstadScreens for use registration_allowed script.

Nikunjsharma0 committed:
Merge remote-tracking branch 'origin/invite_users' into learningspace-invite

# Conflicts:
#	app-ktor-server/.gitignore
#	app-ktor-server/src/main/kotlin/com/ustadmobile/lib/rest/UmRestApplication.kt
#	app-react/src/jsMain/kotlin/com/ustadmobile/entities/UstadScreen.kt
#	core/src/commonMain/kotlin/com/ustadmobile/core/impl/di/CommonDomainModule.kt
#	core/src/commonMain/resources/MR/base/strings.xml
#	lib-database/schemas/com.ustadmobile.core.db.UmAppDatabase/195.json
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/UmAppDatabase.kt
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/ext/UmAppDatabaseMigrations.kt

Nikunjsharma0 committed:
invite branch merged

Nikunjsharma0 committed:
invite bug fix

mikedawson committed:
Work-in-progress migration of SystemDb to using SQLdelight.

Multiple Door databases (e.g. UmAppDatabase and SystemDb) on Kotlin/JS failed: hence need to migrate. Web client will not need a local database.

Nikunjsharma0 committed:
invite bug fix

Nikunjsharma0 committed:
ProcessInviteUseCaseTest issue fix

Nikunjsharma0 committed:
ParseInviteUseCaseTest issue fix

poojaustad committed:
Updated test descriptions to reflect changes

poojaustad committed:
Updated test descriptions to reflect changes

Nikunjsharma0 committed:
Merge branch 'dev-simplified-onboarding-learningspacemgr' into learningspace-invite

poojaustad committed:
Updated test descriptions to reflect changes

Nikunjsharma0 committed:
Merge branch 'learningspace-invite' into dev-simplified-onboarding

# Conflicts:
#	app-android/src/main/java/com/ustadmobile/port/android/impl/UstadApp.kt
#	app-android/src/main/java/com/ustadmobile/port/android/view/AbstractAppActivity.kt
#	lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/login/LoginScreen.kt

mikedawson committed:
Rename lib-systemdb-model to lib-systemdb-core (because it contains common repository interfaces as well).
Update libui compose screens to use lib-systemdb-core classes.

mikedawson committed:
Merge branch 'dev-simplified-onboarding-learningspacemgr' into dev-simplified-onboarding-learningspacemgr-sqd

Nikunjsharma0 committed:
invite user merged

mikedawson committed:
Modularize XXHash

XXHash functionality is needed for systemdb module to convert between network/database entities. Will likely be needed in other modules.

mikedawson committed:
Handle showing list of learning spaces on desktop using SQLDelight based implementation.

mikedawson committed:
Remove explicit dependsOn for lib-systemdb-core and lib-systemdb-sqldelight such that default hierarchy is applied.

mikedawson committed:
Merge remote-tracking branch 'origin/dev-simplified-onboarding' into dev-simplified-onboarding-learningspacemgr-sqd

# Conflicts:
#	app-android/src/main/java/com/ustadmobile/port/android/impl/UstadApp.kt

mikedawson committed:
Fix LearningSpaceServerRepo add function to add learningspace to in memory cache.

mikedawson committed:
Handle listing learning spaces on Android using SQLDelight based systemdb.

mikedawson committed:
Update app-react to use SQLDelight based learning space list.

mikedawson committed:
Merge pull request #991 from UstadMobile/test-simplified-onboarding-testdescription

Test simplified onboarding

mikedawson committed:
Merge remote-tracking branch 'origin/dev-simplified-onboarding' into dev-simplified-onboarding-learningspacemgr-sqd

mikedawson committed:
Merge pull request #967 from UstadMobile/dev-simplified-onboarding-cypress-code-update

Cypress code update

mikedawson committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-learningspacemgr-sqd

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

mikedawson committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-learningspacemgr-sqd

mikedawson committed:
Remove obsolete server Google API key config
Update ktor database init to avoid throwing exception when running via source. See https://github.com/UstadMobile/UstadMobile/issues/994

mikedawson committed:
Fix Cypress 004-005 - after student clicks submit needed to ensure that submission was saved before moving on.

mikedawson committed:
Refactor libsystemdb to lib-centralappconfigdb and update docs.

mikedawson committed:
Merge pull request #995 from UstadMobile/dev-simplified-onboarding-learningspacemgr-sqd

Migrate central app config db to SQLDelight

Nikunjsharma0 committed:
commit

mikedawson committed:
Fix UmAppDatabaseMigration handling of adding ClazzInvite table.
Update CreateLearningSpaceUseCase to create new admin only if not already existing.

Nikunjsharma0 committed:
some class name change and coding style commit

Nikunjsharma0 committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-clean

Nikunjsharma0 committed:
preferImmediatelyAvailableCredentials added

mikedawson committed:
Merge pull request #998 from UstadMobile/dev-simplified-onboarding-clean

Dev simplified onboarding clean

Nikunjsharma0 committed:
saving password in password manager

Nikunjsharma0 committed:
LoginWithPasskeyUseCase.kt changed to GetCredentialUseCase

mikedawson committed:
Fix jvm integration tests to use find and use a random available port instead of using hardcoded default ports.

Nikunjsharma0 committed:
Merge remote-tracking branch 'refs/remotes/origin/dev-simplified-onboarding' into dev-simplified-onboarding-savepassword

Nikunjsharma0 committed:
getting domain from learning space and moving impl to core

mikedawson committed:
Merge pull request #1000 from UstadMobile/dev-simplified-onboarding-savepassword

saving password in password manager

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

# Conflicts:
#	testserver-controller/src/main/java/com/ustadmobile/test/http/TestServerController.kt

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

mikedawson committed:
Fix ServerAppMain setting config file. Set using the default ustad-server.conf file if it exists automatically.

Nikunjsharma0 committed:
username , the text remains in chip field and Email notification subject line is blank fixed

Nikunjsharma0 committed:
commit

mikedawson committed:
Minor invite tidy up.

mikedawson committed:
Merge pull request #1014 from UstadMobile/invite-ui-new

Invite UI new

mikedawson committed:
Fix ParseInviteUseCaseTest

mikedawson committed:
Workaround for TetsServerController compilation.

mikedawson committed:
Fix ServerAppMain to pass config prop using -config= property correctly.
Fix ClassInviteViewModel to redirect user if they are not currently logged in.

Nikunjsharma0 committed:
when invitation sent pop back and sending invitation even  if text present in textfield not created chip

mikedawson committed:
Merge pull request #1018 from UstadMobile/invite-ui-enchanced

invitation ui enhanced

mikedawson committed:
Update invite related files to follow naming conventions.

mikedawson committed:
Improvements to ClazzInvite system:

1) Check invite when redeem screen is opened to ensure it is valid and redeemable
2) Show error messages (if any)
3) Fix queries to ensure updates change last modified time and therefor would sync to server.

Nikunjsharma0 committed:
resend invite added

Nikunjsharma0 committed:
Merge branch 'dev-simplified-onboarding' into invite-ui-resend-invite

# Conflicts:
#	app-react/src/jsMain/kotlin/com/ustadmobile/entities/UstadScreen.kt
#	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/clazz/inviteredeem/ClazzInviteRedeemViewModel.kt
#	lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/app/AppNavHost.kt
#	lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/clazz/inviteviacontact/PendingInviteScreen.kt

Nikunjsharma0 committed:
pending invite list showing in ClazzMemberListScreen

Nikunjsharma0 committed:
Merge branch 'dev-simplified-onboarding' into invite-ui-resend-invite

# Conflicts:
#	core/src/commonMain/kotlin/com/ustadmobile/core/domain/invite/ClazzInviteRedeemUseCase.kt
#	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/clazz/inviteredeem/ClazzInviteRedeemViewModel.kt
#	core/src/commonMain/resources/MR/base/strings.xml
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/ClazzInviteDao.kt

Nikunjsharma0 committed:
simplified onboarding changes merged

Nikunjsharma0 committed:
SendEmailUseCase refracted as interface and implemented real and mock email sender

Nikunjsharma0 committed:
e sign-up view model to handle NEXT argument similarly to Login view model

mikedawson committed:
Initial merge of latest primary updates (primraily changes to run end-to-end tests in parallel / faster) into dev-simplified-onboarding.

All test spec conflicts were handled using 'accept theirs (primary)'

Nikunjsharma0 committed:
useMockEmail set false in appconfig

mikedawson committed:
Update testserver-controller after primary merge to create learning space as required on dev-simplified-onboarding.

mikedawson committed:
Fix testserver-controller cypress run: disable jsDevServer when starting server.

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding-primary-merge

poojaustad committed:
maestro and cypress updates

poojaustad committed:
Merge remote-tracking branch 'origin/dev-simplified-onboarding-primary-merge' into dev-simplified-onboarding-primary-merge

poojaustad committed:
maestro test fix

Nikunjsharma0 committed:
passkey signin added in AddAccountSelectNewOrExisting

poojaustad committed:
test descriptions

poojaustad committed:
test descriptions

mikedawson committed:
Merge pull request #1041 from UstadMobile/dev-simplified-onboarding-primary-merge

Dev simplified onboarding primary merge

mikedawson committed:
Merge pull request #1027 from UstadMobile/invite-ui-resend-invite

Invite UI resend invite

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

mikedawson committed:
Refactoring to ensure that newly added screens/viewmodels accurately follow naming patterns.

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

mikedawson committed:
Improve ClazzInvite handling of keyboard hiding.

Previous version used the ui state in a way that did not follow coding patterns.

Invite view model needs to be able to explicitly hide the software keyboard.

Fix UmRestApplication to ensure data directory is created if needed.

mikedawson committed:
Invitation improvements

1) ClazzInviteViaContactScreen on web: fix handling of backspace. Remove redundant extra state variable. Add key on react component.
2) ClazzInviteViaContactScreen on Jetpack compose: minor documentation improvements and handle chip removal using list (just in case multiple chips are removed, avoid multiple state updates from one event)
3) SendInviteUseCase is an interface with an implementation for the client (which sends the request to the server) and a server implementation which actually stores the invitations in the database and sends emails etc.
4) Remove CheckContactTypeUseCase - this was really duplicating the logic already contained in ParseInviteUseCase. Parse invite use case was using a string and splitting, CheckContactTypeUseCase was using a list of strings. No need to have two with essentially identical logic.
5) Parse invite use case now accepts commas and semicolons as separators

mikedawson reviewed (changes requested):
Send invitation needs to handle the edge case if/when a contact has already been invited (extend the expiration, resend). 

Resend should use the existing send invitation use case - there is no need to repeat the same logic

Nikunjsharma0 committed:
revoke passkey added

Nikunjsharma0 committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-manage-passkey

# Conflicts:
#	core/src/commonMain/resources/MR/base/strings.xml

Nikunjsharma0 committed:
for resend using existing send invitation use case

Nikunjsharma0 committed:
Resend invite use cases removed

Nikunjsharma0 committed:
fix issue 1050

poojaustad committed:
cypress test-005_004 external link test fix for simplified onboarding branch

mikedawson committed:
Merge pull request #1064 from UstadMobile/dev-simplified-onboarding-external-link-test-fix

cypress test-005_004 external link test fix for simplified onboarding…

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

# Conflicts:
#	lib-database/schemas/com.ustadmobile.core.db.UmAppDatabase/204.json
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/UmAppDatabase.kt
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/ext/UmAppDatabaseMigrations.kt

mikedawson committed:
Merge migrations.

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

# Conflicts:
#	app-react/src/jsMain/kotlin/com/ustadmobile/entities/UstadScreen.kt
#	core/src/commonMain/kotlin/com/ustadmobile/core/impl/di/CommonDomainModule.kt
#	lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/UmAppDatabase.kt
#	lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/app/AppNavHost.kt
#	test-end-to-end/android-maestro/e2e-tests/003_003_add_existing_content_in_library_as_block.yaml
#	test-end-to-end/android-maestro/e2e-tests/003_005_add_new_content_block_inside_course.yaml
#	test-end-to-end/test-descriptions/001_001_admin_can_add_content_test_description.md
#	test-end-to-end/test-descriptions/003_003_add_existing_content_in_library_as_block_test_description.md
#	test-end-to-end/test-descriptions/003_005_add_new_content_block_inside_course_test_description.md

mikedawson committed:
Fix compilation after updated primary merge (including attempt list).

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

poojaustad committed:
updated e2e tests for attemptlist test updates

poojaustad committed:
updated e2e tests for attemptlist test updates

poojaustad committed:
updated cypress test

mikedawson committed:
Tidyup handling of using saved passwords.

mikedawson committed:
Tidyup and refactoring of the signup/registration flow and passkey handling.

mikedawson committed:
Merge pull request #1067 from UstadMobile/dev-simplified-onboarding-attemptlst-test

updated e2e tests for attemptlist test updates

mikedawson committed:
Add binding for CheckRegistrationAllowedUseCase

mikedawson committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-manage-passkey

# Conflicts:
#	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt

mikedawson committed:
Fix merge

mikedawson committed:
Fix merge

mikedawson committed:
Merge pull request #1055 from UstadMobile/dev-simplified-onboarding-manage-passkey

Dev simplified onboarding manage passkey

mikedawson committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-invite-fixes

mikedawson committed:
Merge pull request #1054 from UstadMobile/dev-simplified-onboarding-invite-fixes

Dev simplified onboarding invite fixes

mikedawson committed:
Merge branch 'primary' into dev-simplified-onboarding

mikedawson committed:
Fixes for passkey creation and additional documentation

mikedawson committed:
Refactor passkey and credential handling:
1) Create use case for CreateCredentialUsernameUseCase and ParseCredentialUsernameUseCase - this is to be used both when saving a username and password to a password manager and when creating a passkey.
2) Update SystemUrlConfig - remove passkeyRpId. The passkey rpId is always the hostname of the SystemUrlConfig.systemBaseUrl .
3) Update AndroidManifest.xml to use SystemUrlConfig.systemBaseUrl in intent-filter for links

mikedawson committed:
Fix DomainDiModuleJs compilation

mikedawson committed:
Fix DesktopDomainDiModule compilation

poojaustad committed:
Updated button name - Manage account in cypress test

poojaustad committed:
Updated button name - Manage account in maestro test

poojaustad committed:
Manage account in maestro test 007_004

poojaustad committed:
Manage account in maestro test 001_002

mikedawson committed:
Merge pull request #1082 from UstadMobile/dev-simplified-onboarding-manage-account-button-update

Updated button name - Manage account in cypress test

mikedawson committed:
Update passkey creation:
1) Use typed kotlinx serialization classes for passkey creation
2) Make passkey creation domain use cases bound to scope (so they understand what learning space they are connected to, and generate a username using CreateCredentialUsernameUseCase accordingly.
3) Update SignUpViewModel to use loadEntity function.

mikedawson committed:
Merge remote-tracking branch 'origin/dev-simplified-onboarding' into dev-simplified-onboarding

mikedawson committed:
Add learning space url template support to testserver controller to enable maestro end to end tests to use https.

Nikunjsharma0 committed:
username suggestions added

Nikunjsharma0 committed:
Kotlinx Serialization Classes for WebAuthn Passkey Response

Nikunjsharma0 committed:
changes in GetCredentialUseCaseImpl

mikedawson committed:
Fix testServerController running Cypress: changing the order of adding to the running server list broke the reverse proxy interceptor.

Nikunjsharma0 committed:
CreatePasskey use case should be returning the webauthn response
Creating a passkey - use case parameter should be only the username

Nikunjsharma0 committed:
added UsernameSuggestionUseCase

Nikunjsharma0 committed:
spec added

mikedawson committed:
Refactor passkey rpId and user id handling:

Update CreatePublicKeyCredentialCreationOptionsJsonUseCase - set the user handle (user id) as passkeyUid@learningspace - as per the spec avoiding the use of any personal information, but including the server url.

Remove rpId from PersonPasskey (this is system wide, not a per entity variable).

mikedawson committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-username-into-signup

Nikunjsharma0 committed:
changed the name of PasskeyWebAuthNResponse according to spec
change in the flow of signin with passkey

mikedawson committed:
Move webauthn model into model package.

Nikunjsharma0 committed:
user handle encoding decoding added with test

Nikunjsharma0 committed:
clean up encoding decoding

mikedawson committed:
Fix SavePersonPasskeyUseCase compilation

Nikunjsharma0 committed:
username issue fixed for web

mikedawson committed:
Fix AbstractAppActivity compilation and remove CreatePasskeyRequestJsonUseCase from desktop and web DI.

Nikunjsharma0 committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-username-into-signup

# Conflicts:
#	app-android/src/main/java/com/ustadmobile/port/android/view/AbstractAppActivity.kt
#	app-desktop/src/main/java/com/ustadmobile/port/desktop/DesktopDomainDiModule.kt
#	core/src/androidMain/kotlin/com/ustadmobile/core/domain/credentials/passkey/GetCredentialUseCaseImpl.kt
#	core/src/commonMain/kotlin/com/ustadmobile/core/domain/credentials/SavePersonPasskeyUseCase.kt

mikedawson committed:
Minor tidyup - remove commented out code etc.

mikedawson committed:
Remove unused json codegen dependency.

mikedawson committed:
Merge pull request #1084 from UstadMobile/dev-simplified-onboarding-username-into-signup

username suggestions added

mikedawson committed:
Fix ustad-server.conf in distribution version to avoid setting an incorrect .well-known directory.

mikedawson committed:
Add documentation on checking asset links.

mikedawson committed:
Update documentation on checking asset links.

poojaustad committed:
invite user e2e and username test added to cypress tests

poojaustad committed:
invite user e2e and username test added to maestro tests

Nikunjsharma0 committed:
google password manager issues fixed

mikedawson committed:
Merge pull request #1102 from UstadMobile/dev-simplified-onboarding-google-password-mngr-fixes

google password manager issues fixed

poojaustad committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-invite-user-e2e

Nikunjsharma0 committed:
google password manager signing in issues fixed
for github issue
https://github.com/UstadMobile/UstadMobile/issues/1103

poojaustad committed:
maestro - stopserver uncommented

Nikunjsharma0 committed:
SignUpScreen bug fix for issue
 Create account screen -Error message persists after all mandatory fields are completed #1100

poojaustad committed:
Merge remote-tracking branch 'origin/dev-simplified-onboarding-signup-issuefix' into dev-simplified-onboarding-invite-user-e2e

poojaustad committed:
updated cypress - mandatory field check

poojaustad committed:
updated maestro - added max tries on downloads

poojaustad committed:
maestro - 007_007b_admin_send_group_invitation_to_join_course_via_email.yaml moved to pending folder

mikedawson committed:
Merge pull request #1104 from UstadMobile/dev-simplified-onboarding-passkey-signin-fixes

google password manager signing in issues fixed

mikedawson committed:
Merge pull request #1106 from UstadMobile/dev-simplified-onboarding-signup-issuefix

 SignUpScreen bug fix for issue

mikedawson committed:
Remove obsolete CreatePasskeyRequestJsonUseCase
Prevent error being shown when there is no saved credential available.
Tidyup passkey documentation on user handle use cases.

mikedawson committed:
Merge branch 'dev-simplified-onboarding' into dev-simplified-onboarding-invite-user-e2e

mikedawson committed:
Revert changes to application.conf and update ServerRunner to handle useMockEmail

Revert change to application.conf that enabled useMockEmail by default.
Update testserver-controller ServerRunner to set useMockEmail parameter

mikedawson committed:
Merge pull request #1101 from UstadMobile/dev-simplified-onboarding-invite-user-e2e

invite user e2e and username test added to cypress tests

mikedawson committed:
Merge primary into dev-simplified-onboarding.

mikedawson committed:
Fix compilation issues after merging primary into dev-simplifed-onboarding

mikedawson committed:
Delete obsolete scripts and wget test logs.

mikedawson reviewed (approved):
Passkey login is working, and this branch is now functionally ahead of primary.
